test/container/detail/random_access_iterator_test.cpp

```diff
@@ -0,0 +1,480 @@
+//       documentation and/or other materials provided with the distribution.
+//     * Neither the name of Knut Reinert or the FU Berlin nor the names of
+//       its contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+
+/*!\file container/detail/random_access_iterator_test.hpp
+ * \brief Random access iterator for const and non const containers.
+ * \author Marie Hoffmann <marie.hoffmann AT fu-berlin.de>
+ * \ingroup iterator test
+ */
+
+#include <seqan3/container/detail/random_access_iterator.hpp>
+
+#include <gtest/gtest.h>
+#include <sstream>
+#include <vector>
+
+class random_access_iterator_test_fixture: public ::testing::Test {
+protected:
+
+   std::vector<uint8_t> v_empty;
+   const std::vector<uint8_t> v_const_empty{};
+   std::vector<uint8_t> *v, *v2, *v3, *v4, *w, *w2;
+   const std::vector<uint8_t> *v_const, *v2_const, *v3_const, *v4_const, *w_const, *w2_const;
+   std::array<long int, 3> a, b;
+   const std::array<long int, 3> a_const[3] = {11, 22, 33}; //, b_const;
+
+   virtual void SetUp( ) {
+       // code here will execute just before the test ensues
+       v = new std::vector<uint8_t>                 {'a', 't'};
+       v_const = new const std::vector<uint8_t>     {'a', 't'};
+       v2 = new std::vector<uint8_t>                {'a', 'u'};
+       v2_const = new const std::vector<uint8_t>    {'a', 'u'};
+       v3 = new std::vector<uint8_t>                {'a', 't', 'z'};
+       v3_const = new const std::vector<uint8_t>    {'a', 't', 'z'};
+       v4 = new std::vector<uint8_t>                {'a', 'u', 'v', 'w', 'x'};
+       v4_const = new const std::vector<uint8_t>    {'a', 'u', 'v', 'w', 'x'};
+       w = new std::vector<uint8_t>                 {'c', 't'};
+       w_const = new const std::vector<uint8_t>     {'c', 't'};
+       w2 = new std::vector<uint8_t>                {'b', 'v'};
+       w2_const = new const std::vector<uint8_t>    {'b', 'v'};
+       a = {11, 22, 33};
+       //a_const = {11, 22, 33};
+   }
```

Comment:
Why are you using pointers and new, delete here? Couldn't you just use regular variables with in-place initialisation (as with `v_const_empty` for example)?